tasks/update-cnb-dependency/dependencies.go

```diff
@@ -22,14 +22,25 @@ type Dependency struct {
 	Version      string   `toml:"version"`
 }
 
-func (deps Dependencies) MergeWith(newDeps Dependencies) (Dependencies, error) {
+func (deps Dependencies) Update(flags Flags, dep Dependency, depsToAdd Dependencies) (Dependencies, error) {
```

Comment:
I'd pass the `versionLine` and `versionsToKeep` in directly instead of the `flags` struct, it makes it more clear exactly what this function needs in order to do its work.

Reply:
Added as params